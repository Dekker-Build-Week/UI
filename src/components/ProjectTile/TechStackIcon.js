import React from 'react';
import PropTypes from 'prop-types';

import { makeStyles } from "@material-ui/core/styles";
import techStackstyles from "assets/jss/material-dashboard-react/components/techStackStyle.js";

const useStyles = makeStyles(techStackstyles);

const TechStackIcon = (props) => {
    const classes = useStyles();

    const {techStack} = props;

    return (
        <li className = {classes.techList}>
            <img src = {techStack} alt = "tech" className = {classes.techStack} />
        </li>
    );
}

TechStackIcon.propTypes = {
    TechStack : PropTypes.object.isRequired
}

export default TechStackIcon;
