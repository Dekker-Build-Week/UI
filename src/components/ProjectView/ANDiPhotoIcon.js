import React from 'react';
import PropTypes from 'prop-types';

import { makeStyles } from "@material-ui/core/styles";
import andiPhotoStyles from "assets/jss/material-dashboard-react/components/andiPhotoProjectStyle.js";

const useStyles = makeStyles(andiPhotoStyles);

const ANDiPhotoIcon = (props) => {
    const classes = useStyles();

    const { ANDiPhoto } = props;

    return (
        <li className = {classes.ANDiList}>
            <img src = {ANDiPhoto} alt = "ANDi" className = {classes.ANDiPhoto} />
        </li>
    );
}

ANDiPhotoIcon.propTypes = {
    ANDiPhoto : PropTypes.string.isRequired
}

export default ANDiPhotoIcon;
