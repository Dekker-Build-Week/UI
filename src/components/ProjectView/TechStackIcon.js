import React from 'react';
import PropTypes from 'prop-types';

import { makeStyles } from "@material-ui/core/styles";
import techStackstyles from "assets/jss/material-dashboard-react/components/techStackProjectStyle.js";

const useStyles = makeStyles(techStackstyles);

const TechStackIcon = (props) => {
    const classes = useStyles();

    const {techStack} = props;

    return (
        <li className = {classes.techList}>
            <img src = {techStack} alt = "tech" className = {classes.techStack} />
        </li>
    );
}

TechStackIcon.propTypes = {
    techStack : PropTypes.string.isRequired
}

export default TechStackIcon;
