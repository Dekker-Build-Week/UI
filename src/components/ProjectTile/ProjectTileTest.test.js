import React from 'react';

import ANDiPhotoIcon from './ANDiPhotoIcon';
import TechStackIcon from './TechStackIcon';
import renderer from 'react-test-renderer';

describe('ProjectTile test suite', () => {
    it('Renders the ANDiPhotoIcon correctly', () => {
        const tree = renderer
            .create(<ANDiPhotoIcon
                ANDiPhoto = "Photo"></ANDiPhotoIcon>)
            .toJSON();
        
        expect(tree).toMatchSnapshot();
    }),
    it('Renders a TechStackIcon with GridItems correctly', () => {
        const tree = renderer 
            .create(<TechStackIcon
                        techStack = "Stack"></TechStackIcon>)
                .toJSON();

        expect(tree).toMatchSnapshot(); 
    })
});