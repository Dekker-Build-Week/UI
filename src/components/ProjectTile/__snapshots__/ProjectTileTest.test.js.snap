// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectTile test suite Renders a TechStackIcon with GridItems correctly 1`] = `
<li
  className="makeStyles-techList-4"
>
  <img
    alt="tech"
    className="makeStyles-techStack-3"
    src="Stack"
  />
</li>
`;

exports[`ProjectTile test suite Renders the ANDiPhotoIcon correctly 1`] = `
<li
  className="makeStyles-ANDiList-2"
>
  <img
    alt="ANDi"
    className="makeStyles-ANDiPhoto-1"
    src="Photo"
  />
</li>
`;
